src/pybind/mgr/dashboard/frontend/src/app/shared/directives/update-scope.directive.ts

```diff
@@ -0,0 +1,34 @@
+import { AfterViewInit, Directive, ElementRef, Input, Optional } from '@angular/core';
+import { Permissions } from '../models/permissions';
+import { AuthStorageService } from '../services/auth-storage.service';
+import { PermissionScopeDirective } from './permission-scope.directive';
+
+@Directive({
+  selector: 'input:not([cdNoUpdateScope]), select:not([cdNoUpdateScope]), [cdUpdateScope]'
+})
+export class UpdateScopeDirective implements AfterViewInit {
+  private permissions: Permissions;
+  private service_name: keyof Permissions;
+
+  @Input() disabled: boolean;
+
+  constructor(
+    @Optional() private scope: PermissionScopeDirective,
+    private authStorageService: AuthStorageService,
+    private elementRef: ElementRef
+  ) {
+    this.permissions = this.authStorageService.getPermissions();
+  }
+
+  ngAfterViewInit() {
+    if (this.scope !== null) {
+      this.service_name = this.scope.cdScope;
+    }
+    if (
+      this.service_name !== undefined &&
```

Comment:
Another assert here? If the Scope directive is there, an scope should also be defined.

Reply:
@epuertat I couldn't understand how assert works in typescript. Could you give me any reference to it?